Add doc for dynamic threadpool settings (#8588)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/index-settings.md

- `cluster.default_number_of_replicas` (Integer): Controls the default number of replicas for indexes in the cluster. The index-level `index.number_of_replicas` setting defaults to this value if not configured. Default is `1`.

- `cluster.thread_pool.<fixed-threadpool>.size` (Integer): Controls the sizes of both the fixed and resizable queue thread pools. Overrides the defaults provided in `opensearch.yml`.

- `cluster.thread_pool.<scaling-threadpool>.max` (Integer): Sets the maximum size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.

- `cluster.thread_pool.<scaling-threadpool>.core` (Integer): Specifies the core size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.

> Before tuning thread pool settings dynamically, note that these are expert-level settings that can potentially destabilize your cluster. Modifying thread pool settings applies the same thread pool size to all nodes, so it's not recommended for clusters with different hardware for the same roles. Similarly, avoid tuning thread pools shared by both data nodes and cluster manager nodes for the same reason. After making these changes, we recommend monitoring your cluster to ensure it remains stable and performs as expected.

Reviewer: ```suggestion
Before tuning thread pool settings dynamically, note that these are expert-level settings that can potentially destabilize your cluster. Modifying thread pool settings applies the same thread pool size to all nodes, so it's not recommended for clusters with different hardware for the same roles. Similarly, avoid tuning thread pools shared by both data nodes and cluster manager nodes. After making these changes, we recommend monitoring your cluster to ensure that it remains stable and performs as expected.
```

- `cluster.default_number_of_replicas` (Integer): Controls the default number of replicas for indexes in the cluster. The index-level `index.number_of_replicas` setting defaults to this value if not configured. Default is `1`.

> - `cluster.thread_pool.<fixed-threadpool>.size` (Integer): Controls the size of fixed and resizable queue threadpool size. Overrides the defaults provided in yml.

Reviewer 2: Do you mean something like this?
```suggestion
- `cluster.thread_pool.<fixed-threadpool>.size` (Integer): Controls the sizes of both the fixed and resizable queue thread pools. Overrides the defaults provided in `opensearch.yml`.
```

> - `cluster.thread_pool.<scaling-threadpool>.max` (Integer): Controls the maximum size of  scaling threadpool. Overrides the defaults provided in yml.

Reviewer 2: ```suggestion
- `cluster.thread_pool.<scaling-threadpool>.max` (Integer): Sets the maximum size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.
```

> - `cluster.thread_pool.<scaling-threadpool>.core` (Integer): Controls the core size of  scaling threadpool. Overrides the defaults provided in yml.

Reviewer 2: ```suggestion
- `cluster.thread_pool.<scaling-threadpool>.core` (Integer): Specifies the core size of the scaling thread pool. Overrides the default provided in `opensearch.yml`.
```

> ### Notes for dynamically tuning Thread Pool Settings

Reviewer 2: ```suggestion
```

> These are expert level settings, which can potentially destabilize your cluster. It is not advised on heterogeneous clusters, as it will make threadpool count homogenous on all the nodes. It is not advised to tune threadpool which are used in both data node and cluster manager as well , due to the same reason. Post tuning these settings,

Author: Addressed .

> These are expert level settings, which can potentially destabilize your cluster. It is not advised on clusters having different types of hardware for same roles, as it will make threadpool count equal on all of them. It is not advised to tune threadpool which are used in both data node and cluster manager as well , due to the same reason. Post tuning these settings,

Reviewer 2: ```suggestion
Before tuning thread pool settings dynamically, note that these are expert-level settings that can potentially destabilize your cluster. Modifying thread pool settings applies the same thread pool size to all nodes, so it’s not recommended for clusters with different hardware for the same roles. Similarly, avoid tuning thread pools shared by both data nodes and cluster manager nodes for the same reason.After making these changes, we recommend monitoring your cluster to ensure it remains stable and performs as expected.
{: .warning}
```

> We recommend monitoring the cluster to ensure it is stable and performing as expected.

Reviewer 2: ```suggestion
```
